src/lib/hash/shake/shake.cpp

```diff
@@ -45,13 +45,13 @@ void SHAKE_128::clear()
 
 void SHAKE_128::add_data(const uint8_t input[], size_t length)
    {
-   m_S_pos = SHA_3::absorb(SHAKE_128_BITRATE, m_S, m_S_pos, input, length);
+   m_S_pos = Keccak_FIPS_generic::absorb(SHAKE_128_BITRATE, m_S, m_S_pos, input, length);
```

Comment:
Referring to the idea of making `Keccak_FIPS_generic` a template (with the capacity being a value template-parameter), this might be written as:

```C++
Keccak_FIPS_128::absorb(m_S, m_S_pos, input, length);
```

which I would find desirable. But as stated there: certainly okay to look at separately as a follow-up.